### PR TITLE
MAYA-129260 - MayaUsd: make building with Ufe mandatory and remove v1 support

### PR DESCRIPTION
**Pull request by seando-adsk**
#### MAYA-129260 - MayaUsd: make building with Ufe mandatory and remove v1 support
I split the work into smaller chunks, thus the 7 commits. It might be easier to review if you look at each commit separately. 

commit 1: 6e409b64393db3463b5c5c5502043b90361867ab: cmake/doc changes to remove optional building with Ufe.
commit 2: f3c633bbd21cafa567a93971015e91b997e793ce: code changes to remove "WANT_UFE_BUILD" conditional.
commit 3: 869f65215ae1fefc211ace24f402784ec0a1a98f: code changes to remove hard-coded ufe runtime id
commit 4: a04265d3bbd915a5e1a9485c5b81a5f1957f5dbe: cmake changes to remove conditional Ufe v2
commit 5: 30de1c17157de1fe101a956ed171b975b03797b8: code changes to remove conditional Ufe v2
commit 6: 03c78aeee30f186969f40eacabc64a1e1b1ad6dc: Replace UFE_NS with "Ufe"
commit 7: e9eb1f9144db01f08a59cfc28c5915322bf688c9: Don't need "UsdTransform3d/UsdTransform3dHandler/ToggleTransform3d". These were hack for Ufe v1 support.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -135,14 +134,11 @@ if (CMAKE_WANT_MATERIALX_BUILD)
...
 
-if(CMAKE_WANT_UFE_BUILD)
-    find_package(UFE QUIET)
-    if(UFE_FOUND)
-        message(STATUS "Building with UFE ${UFE_VERSION} features enabled.")
-    else()
-        message(STATUS "UFE not found. UFE features will be disabled.")
-    endif()
-endif()
+find_package(UFE 0.2.0 REQUIRED)
```
Comment (seando-adsk) [inline CMakeLists.txt:137]: Ufe is not mandatory and we must find at least v2 (for Maya 2022).

File: plugin/adsk/plugin/plugin.cpp [hunk 2]
```diff
@@ -101,52 +87,6 @@ const MTypeId MayaUsdPreviewSurface_typeId(0x58000096);
 const MString MayaUsdPreviewSurface_typeName("usdPreviewSurface");
 const MString MayaUsdPreviewSurface_registrantId("mayaUsdPlugin");
 
-#if defined(WANT_UFE_BUILD) && defined(UFE_V2_FEATURES_AVAILABLE)
-// Keep a reference to the existing USD Transform3d handler, to restore on
-// finalization.
-Ufe::Transform3dHandler::Ptr g_OldTransform3dHandler;
-Ufe::Transform3dHandler::Ptr g_NewTransform3dHandler;
-
-class ToggleTransform3d : public MPxCommand
```
Comment (seando-adsk) [inline plugin/adsk/plugin/plugin.cpp:110]: This command was a hack used to revert to Maya 2020 behavior. We don't support this anymore. That means we no longer need the UsdTransform3dHandler and UsdTransform3d.

File: doc/codingGuidelines.md [hunk 3]
```diff
@@ -227,8 +227,8 @@ Headers should be included in the following order, with each section separated b
 
 #include <string>
 
-#if defined(WANT_UFE_BUILD)
-  #include <ufe/ufe.h>
+#ifdef UFE_V2_FEATURES_AVAILABLE
```
Comment (samuelliu-adsk) [inline doc/codingGuidelines.md:230]: Do we still need the UFE_V2_FEATURES_AVAILABLE flag if we can only build with UFE V2 available?

Comment (seando-adsk) [reply doc/codingGuidelines.md:230]: Ah right, this flag will always be available in ufe.h, but I should have used the V3 one here.

**Review by samuelliu-adsk — APPROVED** [review]

**Pull request merged**